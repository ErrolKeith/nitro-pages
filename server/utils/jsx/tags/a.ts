import { makeJsx } from "./makeJsx";

export interface AnchorProps {
  href: string;
  target: "_blank" | "_parent" | "_self" | "_top";
  className?: string;
}

export interface AnchorConfig {
  props: AnchorProps;
  children: React.ReactElement[] | string;
}
export interface AnchorProps {
  href: string;
  target: "_blank" | "_parent" | "_self" | "_top";
  className?: string;
}

export interface AnchorConfig {
  props: AnchorProps;
  children: React.ReactElement[] | string;
}

export function makeAnchor(config: AnchorConfig) {
  return makeJsx("a", config.props, config.children);
}
