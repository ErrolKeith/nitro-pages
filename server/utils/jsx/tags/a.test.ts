import { it, expect } from "vitest";
import { makeAnchor, AnchorConfig } from "./a";

it("can make an anchor tag", () => {
  const anchorTestConfig: AnchorConfig = {
    props: {
      href: "#",
      target: "_self",
      className: "a1-test",
    },
    children: "intenral link",
  };
  const anchorTest = makeAnchor(anchorTestConfig);
  expect(anchorTest).toMatchSnapshot();
});
