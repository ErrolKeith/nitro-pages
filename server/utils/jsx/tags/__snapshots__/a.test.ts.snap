// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make an anchor tag 1`] = `
<a
  className="a1-test"
  href="#"
  target="_self"
>
  intenral link
</a>
`;
